app/models/contact.rb

```diff
@@ -12,7 +12,14 @@ class Contact < ApplicationRecord
   has_paper_trail
 
   validates :first_name, presence: true
+  validates :surname, presence: true
   validates_date :date_of_birth, allow_nil: true, allow_blank: true
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, if: :email?
```

Comment:
a*@gmail.com is not a valid email but this regex seems to accept it

Reply:
This regex comes from:
https://html.spec.whatwg.org/multipage/input.html#e-mail-state-(type=email)

* is a valid part of an email
https://en.wikipedia.org/wiki/Email_address#Local-part

Reply:
ok